lib/rouge/lexers/make.rb

```diff
@@ -80,22 +107,22 @@ def initialize(opts={})
 
       state :shell do
         # macro interpolation
-        rule %r/\$\(\s*[a-z_]\w*\s*\)/i, Name::Variable
-        # $(shell ...)
-        rule %r/(\$\()(\s*)(shell)(\s+)/m do
+        rule %r/\$[({][\t ]*[a-z_][\w:=%.]*[\t ]*[)}]/i, Name::Variable
+        # function invocation
+        rule %r/(\$[({])([\t ]*)(#{Make.functions.join('|')})([\t ]+)/m do
           groups Name::Function, Text, Name::Builtin, Text
           push :shell_expr
         end
 
         rule(/\\./m) { delegate @shell }
-        stop = /\$\(|\(|\)|\\|$/
+        stop = /\$\(|\$\{|\(|\)|\}|\\|$/
```

Comment:
I realise this mistake (if it is a mistake) is from the earlier lexer but does it make sense to match against the `$` anchor by itself?

Reply:
I tried taking this out and it could no longer lex lines containing `$$`. This is used to include a literal `$` in the shell command to be executed. I confess I'm not exactly clear on the mechanism by which it decided how to colour the `$$`!

Reply:
Geez, this is why regular expressions are evil. I didn't realise that because this regex is always used via interpolation that the `$` would be interpreted literally. OK, it makes sense why it looks the way it does.